### PR TITLE
feat: Create custom wasm for creating did:trustbloc

### LINKED ISSUE #75: Create custom wasm for creating did:trustbloc

### PR DESCRIPTION
**Pull request by fqutishat**
closes #75

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: cmd/user-agent-js-worker/src/useragent.js [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+             * TODO - remove.
+             * @param text
+             * @returns {Promise<Object>}
+             * @private
+             */
+            _echo: async function (text) {
+                return new Promise((resolve, reject) => {
+                    invoke(aw, pending, "test", "echo", {"echo": text}, "_echo() timed out").then(
+                        resp => resolve(resp.echo),
+                        err => reject(new Error("aries: _echo() failed. error: " + err.message))
```
Comment (sudeshrshetty) [inline cmd/user-agent-js-worker/src/useragent.js]: aries?

Comment (fqutishat) [reply cmd/user-agent-js-worker/src/useragent.js]: typo

Comment (fqutishat) [reply cmd/user-agent-js-worker/src/useragent.js]: done

File: cmd/user-agent-js-worker/src/worker-loader-web.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+*/
+
+export function loadWorker(pending, notifications, paths) {
+    const wasmJS = paths.wasmJS
+    const wasm = paths.wasm
+    const workerJS = paths.dir + "/worker-impl-web.js"
+    const worker = new Worker(workerJS + "?wasmJS=" + wasmJS + "&wasm=" + wasm)
+    worker.onmessage = e => {
+        const result = e.data
+        if (result.topic){
```
Comment (sudeshrshetty) [inline cmd/user-agent-js-worker/src/worker-loader-web.js]: we don't need this check if we are not using sending topics from wasm. (notifier mechanism).

File: cmd/user-agent-js-worker/src/worker-loader-web.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+export function loadWorker(pending, notifications, paths) {
```
Comment (sudeshrshetty) [inline cmd/user-agent-js-worker/src/worker-loader-web.js]: we don't need notifications

Comment (fqutishat) [reply cmd/user-agent-js-worker/src/worker-loader-web.js]: I'm using notifications

File: cmd/user-agent-js-worker/src/worker-loader-web.js (hunk 2 above)
Comment (fqutishat) [reply cmd/user-agent-js-worker/src/worker-loader-web.js]: I need it for https://github.com/trustbloc/edge-agent/pull/76/files#diff-62d1e052170f4151cbcabd61f36085e6R291

File: cmd/user-agent-js-worker/src/wasm_exec.js [hunk 4]
```diff
@@ -0,0 +1,533 @@
+// Copyright 2018 The Go Authors. All rights reserved.
```
Comment (llorllale) [inline cmd/user-agent-js-worker/src/wasm_exec.js]: @fqutishat this should not be committed

**Pull request merged**